src/scripts/oneNoteApiBase.ts

```diff
@@ -48,20 +48,21 @@ export class OneNoteApiBase {
 		return apiRootUrl + partialUrl;
 	}
 
-	private makeRequest(url: string, data?: XHRData, contentType?: string): Promise<ResponsePackage<any> | OneNoteApi.RequestError> {
+	private makeRequest(url: string, data?: XHRData, contentType?: string, verb?: string): Promise<ResponsePackage<any> | OneNoteApi.RequestError> {
 		return new Promise((resolve: (responsePackage: ResponsePackage<any>) => void, reject: (error: OneNoteApi.RequestError) => void) => {
 			let request = new XMLHttpRequest();
 
-			let type: string = data ? "POST" : "GET";
+			let type: string = verb ? verb : data ? "POST" : "GET";
```

Comment:
Making httpMethod a required argument would break backwards compatibility right now right? I think when we upgrade it to a required method it should go in a major release.